demo/index_facets.R

```diff
@@ -0,0 +1,25 @@
+# Xapian provides functionality which fascilitates dynamic generation of complete lists of values which feature in matching documents.
+# This example shows how to ensure that the values you wish to use in facets are stored as document values.
+
+library(RXapian)
+
+# preparing arguments for xapian_index()
+dataFile <- system.file("sampleData/100-objects-v1.csv", package="RXapian")
+
+data <- read.csv(dataFile)
+
+db<- tempfile(pattern="RXapianEx-")
+
+id<-c(0)
+
+indexFields<-list(list(index=2,prefix="S"),
+                  list(index=8,prefix="XD"))
+
+valueS<-list(list(name="COLLECTION",slot=c(0)),list(name="MAKER",slot=c(1)))
```

Comment:
And similarly, a brief explanation of this line, so it's clear what values are created out of what. (It'd probably be helpful to split the outer list across multiple lines, to make this a little more readable.)